service/src/utils/cli_parsers/parseUpdateArgs.ts

```diff
@@ -0,0 +1,111 @@
+import { ethers } from "ethers";
+import { exit } from "process";
+import { Web3Interface } from "scripts/web3-interface";
+import { getOnchainProfile } from "../../utils/onchain-data";
+import { archLogger } from "../chalk-theme";
+import { CLI_BAD_UPDATE_PROFILE_ARG, NO_ONCHAIN_PROFILE } from "../exit-codes";
+
+const diggingFee = 'digging-fee';
+const rewrapInterval = 'rewrap-interval';
+const freeBond = 'free-bond';
+
+export async function parseUpdateArgs(web3Interface: Web3Interface): Promise<{
+    minimumDiggingFee: ethers.BigNumber;
+    maximumRewrapInterval: number;
+    freeBond: ethers.BigNumber;
+}> {
+    const argsStr = process.argv.toString().split("--")[1];
+
+    if (!argsStr) {
+        archLogger.error("Missing arguments to update");
+        exit(CLI_BAD_UPDATE_PROFILE_ARG);
+    }
+
+    const args = argsStr.split(",").map(a => a.trim()).filter(a => a !== "");
+
+    const processedArgs: string[] = [];
+
+    const oldProfile = await getOnchainProfile(web3Interface);
+
+    if (!oldProfile.exists) {
+        archLogger.error("Profile is not registered. Use `npm run register` to register a profile for your account. See readme for usage.");
+        exit(NO_ONCHAIN_PROFILE);
+    }
+
+    const updateProfileParams = {
+        diggingFee: oldProfile.minimumDiggingFee,
+        rewrapInterval: oldProfile.maximumRewrapInterval.toNumber(),
```

Comment:
why are these initialized to the oldProfile values? they should be overridden (so doesn't really matter what they are initialized to), but think they should be initialized to same vals in the register script.

Reply:
The whole data structure is passed into the update contract call, so these are keeping the old profile values unless the user specifically adds arguments to update them. Initial values in register script are Zeros.

Reply:
Ah I see -- I assumed both vars were required to call update (min digging fee / rewrap interval), but since they aren't, then makes sense to init them to the current ones.